File: src/main/java/io/github/natanfudge/impl/events/WorldLoadedEvent.java
package io.github.natanfudge.impl.events;

import net.fabricmc.fabric.api.event.Event;
import net.fabricmc.fabric.api.event.EventFactory;

@FunctionalInterface
public interface WorldLoadedEvent {
    Event<WorldLoadedEvent> EVENT = EventFactory.createArrayBacked(WorldLoadedEvent.class, (susbcribers) -> () -> {
        for (WorldLoadedEvent subscriber : susbcribers) {
            subscriber.onWorldLoaded();
        }
    });

    void onWorldLoaded();
}

File: src/main/java/io/github/natanfudge/impl/events/WorldLoadedEvent.kt
//package io.github.natanfudge.impl.events
//
//import io.github.natanfudge.impl.events.WorldLoadedEvent
//import net.fabricmc.fabric.api.event.EventFactory
//
//internal fun interface WorldLoadedEvent {
//    fun onWorldLoaded()
//
//    companion object {
//        val EVENT = EventFactory.createArrayBacked(WorldLoadedEvent::class.java) { susbcribers ->
//            WorldLoadedEvent {
//                for (subscriber in susbcribers) {
//                    subscriber.onWorldLoaded()
//                }
//            }
//        }
//    }
//}